src/features/DebugSession.ts

```diff
@@ -80,6 +80,12 @@ export class DebugSessionFeature implements IFeature {
                     }
                 }
             }
+            else if (config.script) {
+                // In this case, the user has explicitly defined a script path
+                // so make sure to set the cwd to that path if the cwd wasn't
+                // explicitly set
+                config.cwd = config.cwd || config.script;
```

Comment:
Currently I'm not saving the untitled file contents to a file.  Since the language server already has the contents of the untitled file, I just execute the entire file buffer as a script without saving to a file first.  This is also what the ISE does with untitled files when you press F5.  In the future I might want to save to a temp file so that we can set virtual breakpoints in the untitled script, but then we might run into issues for folks with a restrictive ExecutionPolicy set.

Now that I think about it, does it even make sense to try and set the cwd path when running a script in the shared integrated console session?  It's basically resetting the user's current file path.  Definitely makes sense with unique debug sessions, but we're not back to that point yet.

Sticking with the Environment.CurrentDirectory approach for now could be fine until we have more time to come up with a better solution. I'm planning to ship a patch update tomorrow morning to resolve some issues we found while preparing demos for the PS Summit.